### PR TITLE
feat(select-field): Allow rendering custom options

### PR DESCRIPTION
**Pull request by jpfranco**
This update will allow passing a `optionRenderer` function to `BaseSelectField`, which can be used to render custom select options. 

This PR also contains the following fixes:

- Avoid scrolling options into view when hovering with the mouse (will still do this for keyboard actions).
- Clicking on the scrollbar (when overlay is set to `overflow: auto` and has max height) no longer closes the dropdown.
- Changed `scrollIntoView` mode in order to make the transition less jarring.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jpfranco has signed the CLA. Thanks for the pull request!

File: src/components/select-field/BaseSelectField.js [hunk 1]
```diff
@@ -321,12 +330,8 @@ class BaseSelectField extends React.Component<Props, State> {
 
                     this.selectOption(index);
                 },
-                /* preventDefault on mousedown so blur doesn't happen before click */
-                onMouseDown: event => {
```
Comment (jpfranco) [inline src/components/select-field/BaseSelectField.js]: This was causing the issue where clicking on the scrollbar (when overlay has max height) would close the dropdown. Setting the mouse down event on the overlay itself achieves the same result without causing the issue

File: src/components/select-field/BaseSelectField.js [hunk 2]
```diff
@@ -80,20 +83,26 @@ class BaseSelectField extends React.Component<Props, State> {
...
 
     setActiveItemID = (id: ?string) => {
+        const { shouldScrollIntoView } = this.state;
         const itemEl = id ? document.getElementById(id) : null;
-        this.setState({ activeItemID: id });
-        scrollIntoView(itemEl);
+
+        this.setState({ activeItemID: id, shouldScrollIntoView: false });
+
+        if (shouldScrollIntoView) {
```
Comment (priyajeet) [inline src/components/select-field/BaseSelectField.js]: Should this be part of the setState callback?

Comment (jpfranco) [reply src/components/select-field/BaseSelectField.js]: Yeah, I think so. Will update this

**Pull request merged**